core/deployment/src/main/java/io/quarkus/deployment/recording/PropertyUtils.java

```diff
@@ -28,7 +28,12 @@
                 if (i.getName().startsWith("get") && i.getName().length() > 3 && i.getParameterCount() == 0
                         && i.getReturnType() != void.class) {
                     String name = Character.toLowerCase(i.getName().charAt(3)) + i.getName().substring(4);
-                    getters.put(name, i);
+                    Method existingGetter = getters.get(name);
+                    // In some cases the overridden methods from supertypes can also appear in the array (for some reason).
+                    // We want the most specific methods.
+                    if (existingGetter == null || existingGetter.getReturnType().isAssignableFrom(i.getReturnType())) {
+                        getters.put(name, i);
+                    }
```

Comment:
This one change makes the whole CI test much slower, as all modules depend on it.

Since it looks like we'll have to run the tests several more times, might be easier of you submit this particular change as a separate PR first?

Reply:
I'll have a look, though we'd need to get it merged fast in order for the separate PR to be useful... And that separate PR will have the exact same problem of needing to rebuild everything :)

Reply:
Yes but as soon as you send it I will approve it :)

Reply:
https://github.com/quarkusio/quarkus/pull/17279